src/compiler/crystal/command.cr

```diff
@@ -425,26 +425,30 @@ class Crystal::Command
     end
   end
 
-  private def setup_simple_compiler_options(compiler, opts)
-    opts.on("-d", "--debug", "Add symbolic debug info") do
-      compiler.debug = true
+  private def setup_simple_compiler_options(compiler, opts, no_codegen = false)
```

Comment:
The flag should better be named positively: `codegen = true`. This resolves the double negation from `unless no_codegen` to `if codegen`.